lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2306,7 +2306,7 @@ static bool validateSwiftModuleFileArgumentAndAdd(const std::string &swiftModule
     Diags.diagnose(SourceLoc(), diag::error_bad_module_name, moduleName, false);
     return true;
   }
-  ExplicitSwiftModuleInputs.emplace_back(std::make_pair(moduleName, modulePath));
+  ExplicitSwiftModuleInputs.insert(std::make_pair(moduleName, modulePath));
```

Comment:
What is the rule for this argument? Usually it is last argument wins, not the first one. I guess current implementation is also not clear which one wins.

Reply:
Last argument wins. Added a warning to canonicalize this and made it more explicit in the code.